lib/natalie/compiler2/instructions/push_rel_instruction.rb

```diff
@@ -0,0 +1,24 @@
+require_relative './base_instruction'
+
+module Natalie
+  class Compiler2
+    class PushRelInstruction < BaseInstruction
+      def initialize(index)
+        @index = index
+      end
+      def to_s
+        "push st[#{@index}]"
```

Comment:
Let's call this `push_rel`:

```suggestion
        "push_rel #{@index}"
```

Reply:
On second thought, I wonder if we should call this instruction `dup_rel` 🤔

Reply:
Maybe just make it `dup`?
current `dup` is just the `push st[0]` case
So
`dup` -> `dup 0`
`push st[N]` -> `dup N`

---
btw I just used a notation similar to x87 indexing, because that's what I knew

Reply:
I don't know why, but `dup 0` feels weird to me. I'd be ok with having `dup` and `dup_rel <index>`.